src/NuGet.Core/NuGet.Protocol.Core.v3/HttpSource/HttpSource.cs

```diff
@@ -317,6 +317,11 @@ public Task<JObject> GetJObjectAsync(Uri uri, ILogger log, CancellationToken tok
                             await UpdateHttpClient(promptCredentials);
                             continue;
                         }
+
+                        // null means cancelled by user
+                        // block subsequent attempts to annoy user with prompts
+                        _authRetries = HttpHandlerResourceV3Provider.MaxAuthRetries;
```

Comment:
The motivation behind the fix was addressing a regression from 3.3:
- 3 retries of credential prompts when user hits cancel
- Failing to authenticate means restarting visual studio, given that vsts plans to deprecate credentials rather fast, I don’t think that is acceptable